src/pages/Profile.tsx

```diff
@@ -112,30 +115,38 @@ const Profile = () => {
     username: "",
     followerCount: 0,
     followingCount: 0,
+    followStatus: false,
   });
 
   useEffect(() => {
+    setLoading(true);
```

Comment:
I don't think it makes a difference, but can we move the `setLoading` to inside of the `fetchProfileContents` function? I think it makes the order slightly more clear. So maybe do `setLoading(true)` before the try catch and do `setLoading(false)` in a `finally` block